util/file_reader_writer.cc

```diff
@@ -512,19 +513,24 @@ class ReadaheadRandomAccessFile : public RandomAccessFile {
         return Status::OK();
       }
     }
+    size_t aligned_offset = TruncateToPageBoundary(alignment_, offset + copied);
     Slice readahead_result;
-    Status s = file_->Read(offset + copied, readahead_size_, &readahead_result,
-      buffer_.get());
+    Status s = file_->Read(aligned_offset, readahead_size_ + alignment_,
+                           &readahead_result, buffer_.get());
```

Comment:
Thank you, now I see your point. Indeed, this code guarantees that memory aligns to a page boundary: https://github.com/facebook/rocksdb/blob/master/util/aligned_buffer.h#L107L109 and I should do the same.
I'll use AlignedBuffer.